### PR TITLE
Implement ReStyle

### PR DESCRIPTION
**Pull request by Shysto**
Related issue #478 
Implement the inversion and the toonification tasks.
[Original repo](https://github.com/yuval-alaluf/restyle-encoder)

**IMPORTANT: Needs ailia update for solving segmentation fault error (cf. Slack)**

### CODE REVIEW

File: generative_adversarial_networks/restyle-encoder/align_crop.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+"""
+
+import argparse
+import numpy as np
+import os
+import PIL
+import PIL.Image
+import scipy
+import scipy.ndimage
+import dlib
```
Comment (kyakuno) [inline generative_adversarial_networks/restyle-encoder/align_crop.py:23]: Is it possible to make it work without dlib? Please refer to the psgan code and get the key points with face_alignment. Make dlib available with the --use_dlib option.
https://github.com/axinc-ai/ailia-models/tree/master/style_transfer/psgan

File: generative_adversarial_networks/restyle-encoder/restyle-encoder.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
+import os
+import sys
+import subprocess
+import time
+
+import numpy as np
+import cv2
+import onnx
```
Comment (kyakuno) [inline generative_adversarial_networks/restyle-encoder/restyle-encoder.py:8]: Load onnx and onnx runtime only if the onnx_runtime option is specified.

File: generative_adversarial_networks/restyle-encoder/align_crop.py (hunk 1 above)
Comment (Shysto) [reply generative_adversarial_networks/restyle-encoder/align_crop.py]: The commit I just pushed should implement the --use_dlib opiton corretly.
I had to modify a bit one file in the psgan project, but it shouldn't have any effect on its original execution.

File: generative_adversarial_networks/restyle-encoder/restyle-encoder.py (hunk 2 above)
Comment (Shysto) [reply generative_adversarial_networks/restyle-encoder/restyle-encoder.py]: Done.

**Review by kyakuno — APPROVED** [review]

**Pull request merged**